src/FeeRouter/RewardReceiver.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.16;
+
+interface IArbSys {
+    function withdrawEth(
+        address destination
+    ) external payable returns (uint256);
+}
+
+contract RewardReceiver {
+    address immutable parentChainTarget;
+    uint256 immutable minDistributionIntervalSeconds;
+    uint256 public nextDistribution;
+
+    constructor(address _parentChainTarget,  uint256 _minDistributionIntervalSeconds) {
+        parentChainTarget = _parentChainTarget;
+        minDistributionIntervalSeconds = _minDistributionIntervalSeconds;
+    }
+
+    receive() external payable {
+        require(block.timestamp >= nextDistribution, "RewardRouter: distributing too soon");
```

Comment:
the receive function should never revert, we can consider to skip the withdrawal instead

Reply:
you thinking so that the other recipients still get paid out?

Reply:
other recipients will always get paid out, it just added the annoyance that the failed transfer would be sent to the owner (and if that also revert, the entire distribution revert)